x/tokenfactory/keeper/keeper_test.go

```diff
@@ -25,6 +25,15 @@ func TestKeeperTestSuite(t *testing.T) {
 	suite.Run(t, new(KeeperTestSuite))
 }
 
+func (suite *KeeperTestSuite) SetupGenesisTest() {
+	suite.Setup()
+
+	// remove module account to ensure initGenesis initializes it on its own
+	moduleAddress := suite.App.AccountKeeper.GetModuleAddress(types.ModuleName)
+	tokenfactoryModuleAccount := suite.App.AccountKeeper.GetAccount(suite.Ctx, moduleAddress)
+	suite.App.AccountKeeper.RemoveAccount(suite.Ctx, tokenfactoryModuleAccount)
```

Comment:
Err, what I meant is that make a new thing in apptesting, that does `suite.Setup` except without calling `InitGenesis`

Reply:
wait so something like this; 

```
**apptesting.go** 

func SetupTest() {
...
s.SetupGenesisTest()
}

func (s *KeeperTestHelper) SetupGenesisTest() {
	accountKeeper := s.App.AccountKeeper

	moduleAddress := accountKeeper.GetModuleAddress(types.ModuleName)
	tokenFactoryModuleAccount := accountKeeper.GetAccount(s.Ctx, moduleAddress)
	accountKeeper.RemoveAccount(s.Ctx, tokenFactoryModuleAccount)
}
```

Reply:
Also just curious, why is `SetupGenesisTest` in Keeper_test not good?

Reply:
No not editing SetupTest. 

Make a new function in apptesting called `SetupTestForInitGenesis`, that does not run the call to InitChain located right here: https://github.com/osmosis-labs/osmosis/blob/5948cbc1951bb3e3c7b8e870ff75c11838b1b289/app/test_helpers.go#L26-L32

Reply:
Just pushed the update [`2bd4c23`](https://github.com/osmosis-labs/osmosis/pull/2513/commits/2bd4c2362cdd58f491ff4185172532d43d927980)

I'm pretty sure this "doesnot run the call to initChain" but the KV Store doesnot seem to be registeed 
err: `suite.go:77: test panicked: kv store with key KVStoreKey{0xc00138eb20, acc} has not been registered in stores`

Reply:
nvm i figured it out! 

should be good now [`4971d1c`](https://github.com/osmosis-labs/osmosis/pull/2513/commits/4971d1c5583b39550d7866f5b0bba7bd6288640a)